lib/puppet/type/dism.rb

```diff
@@ -52,15 +40,4 @@
     #defaultto([0, 3010])
     defaultto([0, 3010, 3010 & 0xFF])
   end
-
-  def munge_boolean(value)
-    case value
-      when true, "true", :true
-        :true
-      when false, "false", :false
-        :false
-      else
-        fail("munge_boolean only takes booleans")
-    end
-  end
```

Comment:
So which one of the three applies? Keep in mind Puppet v3 AND Puppet v4.

~~~puppet
dism { 'something':
 all => "true",
}

dism { 'something':
 all => true,
}

dism { 'something':
 all => :true,
}

~~~